hydra_agent/partial_collection_crawler.py

```diff
@@ -0,0 +1,95 @@
+from requests import get
+
+
+class PartialCollectionCrawler:
+    """
+    Crawler for Crawling Collections. 
+    It can move forwards, backwards or can jump to specific page
+    """
+
+    def __init__(self, response):
+        self.response = response
+        self.base_url = 'http://localhost:8080'
+
+    def initialize_forward(self):
+        """
+        Initializes the crawler to move forwards.
+        :returns: Iterator
+        """
+        view = self.response['view']
+        present_page = view['@id']
+        while True:
+            self.response = get(self.base_url + present_page).json()
+            yield self.response['members']
+            view = self.response['view']
+            print('view')
+            if 'next' in view:
+                present_page = view['next']
+            else:
+                raise StopIteration("No more collection pages")
```

Comment:
Is it a good idea to raise the exception in iterators? Do we really need exceptions here? Is it really an exception if no pages are left? 
Think about all these questions, research a bit, and modify your implementation accordingly.

Reply:
When I saw other APIs, there are generally two approach people take. Either omit the next key or give the message, that one has reached the last page. I have opted the later approach, by adding `pages_ahead` and `pages_behind` bool in response. One can check that to find if he can go ahead or backwards are not.